### PR TITLE
MCLOUD-5684: Pagination Not working - product_list_limit=all

### PR DESCRIPTION
**Pull request by dyushkin**
### Fixed Issues (if relevant)

1. https://jira.corp.magento.com/browse/MCLOUD-5684

### Manual testing scenarios

1. Generate small performance toolkit
2. Reindex catalog with bin/magento indexer:reindex
3. Open PLP, it should contain products

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

releasenotes
Added fix to ensure that all products appear on the category page in _Show all pages_ view.

### CODE REVIEW

File: patches/MCLOUD-5684__pagination_not_working_product_list_limit_=_all__2.3.2.patch [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+         $searchCriteria = $this->builder->create();
+         $searchCriteria->setRequestName($this->searchRequestName);
+         $searchCriteria->setSortOrders($this->orders);
+         $searchCriteria->setCurrentPage($this->currentPage - 1);
++        if ($this->size) {
++            $searchCriteria->setPageSize($this->size);
++        }
+ 
+         return $searchCriteria;
+     }
```
Comment (BaDos) [inline patches/MCLOUD-5684__pagination_not_working_product_list_limit_=_all__2.3.2.patch]: Plz, add a new line

Comment (dyushkin) [reply patches/MCLOUD-5684__pagination_not_working_product_list_limit_=_all__2.3.2.patch]: Done!

**Review by BaDos — DISMISSED** [review]

**Review by oshmyheliuk — DISMISSED** [review]

**Comment by shiftedreality** [comment]
@dyushkin please resolve conflicts

**Comment by andriyShevtsov** [comment]
Approved

**Comment by andriyShevtsov** [comment]
Need to fix conflicts

**Review by oshmyheliuk — APPROVED** [review]

**Comment by dyushkin** [comment]
@shiftedreality @andriyShevtsov conflict resolved

**Review by BaDos — APPROVED** [review]

**Pull request merged**